main.go

```diff
@@ -175,6 +176,28 @@ func main() {
 		os.Exit(1)
 	}
 
+	kubeConfigArgs := kubeConfig.ExecProvider.Args
```

Comment:
This is a very hacky approach to get the cluster-name from KubeConfig. I could not find any other ways to get the clusterName. None of the APIs were in here: https://pkg.go.dev/k8s.io/client-go outputs the cluster name.

Reply:
We are using this as Fallback approch right to determine IpFamily. Customers can always specify using yaml spec ?

Reply:
Yes.

Reply:
This is quite brittle no ? Do we need a fallback ? Is this because of existing deployments in the field ?

Reply:
EKS currently doesn't support dual stack. The IpPreference values like `IPv4_PREFERRED` and `IPv6_PREFERRED` (which are a mix of IPv4 and IPv6) can't be supported as a EKS cluster could either be IPv4 or IPv6. We decided to expose only IPv4 or IPv6 options to the customers. 

Given IpPreference can either be - IPV4_only, IPV6_only and option to set IpPreference is optional. By default, we decided to derive ipFamily the option from underlying EKS cluster. If ipFamily IPV6, set IpPreference to IPV6_only. If ipFamily IPV4, set IpPreference to IPV4_only.

When the option is not set App Mesh uses a mix of IPv4_PREFFERED, IPv6_PREFFERED, and IPv4_ONLY. Therefore we need a default value according to the ipFamily used by the cluster.

Reply:
I guess the question is why is it optional ? If you want to stick with this decision, can you think of a better option that introspecting the command line switch ?